From: Baolin Wang <baolin.wang@linux.alibaba.com>
Subject: mm-shmem-support-large-folio-allocation-for-shmem_replace_folio-fix-fix
Date: Wed, 28 Aug 2024 18:17:18 +0800

fix build without CONFIG_TRANSPARENT_HUGEPAGE

ERROR:root:/usr/bin/ld: mm/shmem.o: in function `shmem_swapin_folio':
shmem.c:(.text+0x12d9): undefined reference to `vma_thp_gfp_mask'
collect2: error: ld returned 1 exit status

Link: https://lkml.kernel.org/r/8c03467c-63b2-43b4-9851-222d4188725c@linux.alibaba.com
Signed-off-by: Baolin Wang <baolin.wang@linux.alibaba.com>
Reported-by: SeongJae Park <sj@kernel.org>, Andrew Morton <akpm@linux-foundation.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/shmem.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/mm/shmem.c~mm-shmem-support-large-folio-allocation-for-shmem_replace_folio-fix-fix
+++ a/mm/shmem.c
@@ -1899,11 +1899,13 @@ static int shmem_replace_folio(struct fo
 	 * limit chance of success by further cpuset and node constraints.
 	 */
 	gfp &= ~GFP_CONSTRAINT_MASK;
+#ifdef CONFIG_TRANSPARENT_HUGEPAGE
 	if (nr_pages > 1) {
 		gfp_t huge_gfp = vma_thp_gfp_mask(vma);
 
 		gfp = limit_gfp_mask(huge_gfp, gfp);
 	}
+#endif
 
 	new = shmem_alloc_folio(gfp, folio_order(old), info, index);
 	if (!new)
_
